### PR TITLE
Dispatch buttons render loading pill when clicked

### PR DESCRIPTION
**Pull request by slj**
Connects #711 

Come to think of it, these changes only work in Dispatch.  If that's an issue, please advise.

### CODE REVIEW

File: client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx [hunk 1]
```diff
@@ -88,15 +88,16 @@ export default class EstablishClaimEmail extends BaseForm {
...
+              name="Cancel"
+              onClick={this.props.handleCancelTask}
+              classNames={["cf-btn-link", "cf-adjacent-buttons"]}
             />
             <Button
-            name="Finish Routing Claim"
-            classNames={["usa-button-primary"]}
-            disabled={!this.state.emailForm.confirmBox.value}
-            onClick={this.props.handleEmailSubmit}
+              loading={this.props.loading}
```
Comment (sharonwarner) [inline client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx]: Loading doesn't make sense as a prop since it changes. We should be able to move this to the state object. Same with EstablishClaimNote.

File: client/app/components/RenderFunctions.jsx [hunk 2]
```diff
@@ -9,58 +9,76 @@ export let loadingSymbolHtml = function() {
...
+              0.87264 0.38497,0.54489 0.46392,1.21465 0.21659,1.83621 l -1.71245,4.30622
+              c -0.0733,0.18525 -0.0733,0.38882 0,0.57339 l 1.71454,4.31097 c 0.24734,
+              0.62224 0.16769,1.29131 -0.21728,1.83689 -0.38568,0.54625 -1.01588,0.87264
+              -1.68591,0.87264"/>
           </svg>
           <svg
-            width="30"
-            height="30"
-            viewBox="0 0 500 500"
+            width="20"
```
Comment (sharonwarner) [inline client/app/components/RenderFunctions.jsx]: There are definitely going to be some merge conflicts here. @mdbenjam has a PR open that also deals with the [loading button](https://github.com/department-of-veterans-affairs/caseflow/pull/1072). Probably worth taking a look at that.

Comment (slj) [inline client/app/components/RenderFunctions.jsx]: These changes are in line with the style guide, so I don't see a problem here, although I will discuss it with @mdbenjam. Thx!

File: client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx (hunk 1 above)
Comment (slj) [inline client/app/containers/EstablishClaimPage/EstablishClaimEmail.jsx]: Got it.  Thx.

File: client/app/components/RenderFunctions.jsx (hunk 2 above)
Comment (mdbenjam) [inline client/app/components/RenderFunctions.jsx]: I've merged my changes in, so it should be fine. You'll probably have to deal with some merge conflicts. I can help you work through them if you'd like!

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx [hunk 3]
```diff
@@ -640,7 +639,6 @@ export default class EstablishClaim extends BaseForm {
             handleSubmit={this.handleFormPageSubmit}
             handleFieldChange={this.handleFieldChange}
             handleBackToDecisionReview={this.handleBackToDecisionReview}
-            loading={loading}
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: Hey, so the reason we've been storing `loading` at the top level component (EstablishClaim), is because the parent component has all the context about whether the attempted action has succeeded or failed (that is, when the spinner should stop and start).

For example, the [review page submit](https://github.com/department-of-veterans-affairs/caseflow/blob/f644b7c4842f73f098b5023ef6af2526a60b2e01/client/app/containers/EstablishClaimPage/EstablishClaim.jsx#L369-L390). Note that after the request succeeds, we set the loading indicator to false. This is especially important when the API request may have failed (and you get an error alert). In this scenario the user needs the `loading` set to false so that they can re-submit.

So the functions that make the API calls, and the `loading` state have to live at the same level. So you could in theory move all the API functions down into the sub-pages. This probably makes sense long term but may be harder for the sake of this PR. Otherwise, you'll need to keep the `loading` state at the top level.

Comment (slj) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: How will the sub page components know about `loading` unless I pass it down?  And if I pass it down, how will it be referenced unless I use something like `this.props.loading`?  I was discouraged from this syntax earlier.

Comment (slj) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: OK I understand where I was going wrong.  Everything is fixed.

File: client/app/containers/CaseWorker/CaseWorkerIndex.jsx [hunk 4]
```diff
@@ -59,11 +76,13 @@ export default class CaseWorkerIndex extends BaseForm {
...
               { availableTasks &&
               <Button
+                app={app}
                 name={buttonText}
-                onClick={this.onClick}
+                onClick={this.establishNextClaim}
                 classNames={["usa-button-primary", "cf-push-right",
                   "cf-button-aligned-with-textfield-right"]}
                 disabled={!availableTasks}
+                loading={this.state && this.state.loading}
```
Comment (joofsh) [inline client/app/containers/CaseWorker/CaseWorkerIndex.jsx]: You shouldn't need to check for state first. State will exist already if you create a constructor function. The constructor gets run before the first render, and allows you to define the initial state with loading false. See EstablishClaim for an example

File: client/app/containers/EstablishClaimPage/EstablishClaimComplete.jsx [hunk 5]
```diff
@@ -75,10 +75,12 @@ export default class EstablishClaimComplete extends React.Component {
       <div className="cf-push-right">
         { availableTasks &&
         <Button
+          app={app}
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimComplete.jsx]: The 'app' property in the Button component makes a lot of sense and looks great. Right here, we are inside the EstablishClaimComplete page so we know we're in dispatch already. We should be able to hardcode `app="dispatch"` here

File: app/views/tasks/assigned_existing_ep.html.erb [hunk 6]
```diff
@@ -5,8 +5,9 @@
     checklist: ['Reviewed decision', 'Assigned existing EP to claim'],
     firstHeader: 'Congratulations!',
     page: 'EstablishClaimComplete',
+    app: 'dispatch',
```
Comment (joofsh) [inline app/views/tasks/assigned_existing_ep.html.erb]: If we hardcode the app (see later comment) we won't need to pass it as a property here

File: client/app/components/Button.jsx [hunk 7]
```diff
@@ -10,8 +9,20 @@ export default class Button extends React.Component {
     }
   }
 
+  loadingClasses = (app, loading) => {
```
Comment (joofsh) [inline client/app/components/Button.jsx]: Looks like app is now a required property. Can we add that to the components PropTypes

File: client/app/containers/CaseWorker/CaseWorkerIndex.jsx [hunk 8]
```diff
@@ -7,7 +7,19 @@ import Button from '../../components/Button';
 import { formatDate } from '../../util/DateUtil';
 
 export default class CaseWorkerIndex extends BaseForm {
-  onClick = () => {
+  buildUserRow = (caseInformation) =>
```
Comment (joofsh) [inline client/app/containers/CaseWorker/CaseWorkerIndex.jsx]: Are these changes needed? I don't see it referenced in this PR

File: client/app/components/Button.jsx (hunk 7 above)
Comment (slj) [inline client/app/components/Button.jsx]: You can actually omit it and a default icon will be used.
